pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -273,3 +274,33 @@ int32_t     get_index(t_ndarray arr, ...)
     va_end(va);
     return (index);
 }
+
+/*
+** convert numpy strides to nd_array strides, and return it in a new array, to
+** avoid the problem of different implementations of strides in numpy and ndarray.
+*/
+int64_t     *numpy_to_ndarray_strides(int64_t *np_strides, int type_size, int nd)
+{
+    int64_t *ndarray_strides;
+
+    ndarray_strides = (int64_t*)malloc(sizeof(int64_t) * nd);
+    for (int i = 0; i < nd; i++)
+        ndarray_strides[i] = np_strides[i] / type_size;
+    return ndarray_strides;
+
+}
+
+/*
+** copy numpy shape to nd_array shape, and return it in a new array, to
+** avoid the problem of plateform depending architecture

```

Comment:
Typo: plateform -> platform
Typo: depending -> dependent

And what do you mean by "platform-dependent architecture" anyway?

Reply:
I mean different system architecture 32-bit, 64-bit ...

Reply:
Numpy shape is not saved in a fixed length type

Reply:
Thanks! Please improve the explanation in this comment...